src/Components/Endpoints/src/Builder/ComponentTypeMetadata.cs

```diff
@@ -20,9 +20,25 @@ public ComponentTypeMetadata([DynamicallyAccessedMembers(Component)] Type compon
         Type = componentType;
     }
 
+    /// <summary>
+    /// Initializes a new instance of <see cref="ComponentTypeMetadata"/>.
+    /// </summary>
+    /// <param name="componentType">The component type.</param>
+    /// <param name="isStaticPage">A flag indicating whether the page's route is declared as static.</param>
+    public ComponentTypeMetadata([DynamicallyAccessedMembers(Component)] Type componentType, bool isStaticPage)
+        : this(componentType)
+    {
+        IsStaticRoute = isStaticPage;
```

Comment:
Nit: We can discuss this in API review, but is there a particular reason for naming the parameter `isStaticPage` while the property is name `IsStaticRoute`?

Reply:
Yeah good point. It should probably be `IsStaticRoute` all the way.

Reply:
And more broadly as @javiercn has already suggested, the whole name `@staticpage` could do with change/clarification. We want a name that means "suppress global interactivity for this page" but `@suppressglobalinteractivity` doesn't look great!